src/Autocomplete/src/EntityAutocompleterInterface.php

```diff
@@ -18,30 +18,48 @@
 /**
  * Interface for classes that will have an "autocomplete" endpoint exposed.
  *
+ * @template T of object
+ *
  * @method mixed getGroupBy() Return group_by option.
  */
 interface EntityAutocompleterInterface
 {
     /**
      * The fully-qualified entity class this will be autocompleting.
+     *
+     * @return class-string<T>
      */
     public function getEntityClass(): string;
 
     /**
      * Create a query builder that filters for the given "query".
+     *
+     * @param EntityRepository<T> $repository
      */
     public function createFilteredQueryBuilder(EntityRepository $repository, string $query): QueryBuilder;
 
     /**
      * Returns the "choice_label" used to display this entity.
+     *
+     * @param T $entity
      */
     public function getLabel(object $entity): string;
 
     /**
      * Returns the "value" attribute for this entity, usually the id.
+     *
+     * @param T $entity
      */
     public function getValue(object $entity): mixed;
 
+    /**
+     * Returns the autocomplete result, usually the label as "text" and the value as "value".
+     * May be used to expose extra options such as "disabled" so that tom-select's rendering may be customized.
+     *
+     * @param T $entity
+     */
+    public function getResult(object $entity): array;
```

Comment:
Adding a public method to a non-internal interface is a BC break. 
When upgrading to a minor version, users implementing interface will see their code break because they do not implement this method `getResult()`.

Reply:
But you can set this method in comment for future version and add it into the concrete class.

But.. i feel this method a bit odd.. the result is the object. So maybe more getAttributes or getItemAttr (something that would be 100% opt-in, like what's done in the form) ?

Reply:
Oops, you're 100% right! I'm not used to Open Source development 😅 
I fixed it the way @smnandre suggested (just like the `getGroupBy()` method actually).

Reply:
@smnandre Actually the result is not the object: it's an array like:
```
$results[] = [
    'value' => $autocompleter->getValue($entity),
    'text' => $autocompleter->getLabel($entity),
]
```

I named the method after the `$results` variable that was used in the `AutocompleteResultsExecutor`, but if you have a more relevant suggestion, I don't mind changing it 😉

Reply:
Understood! See other suggestions below regarding naming but mostly "meaning" of the new method